gamestonk_terminal/common/behavioural_analysis/reddit_model.py

```diff
@@ -0,0 +1,530 @@
+"""Reddit Model"""
+__docformat__ = "numpy"
+
+from typing import List, Dict, Tuple
+from datetime import datetime, timedelta
+import pandas as pd
+from prawcore.exceptions import ResponseException
+from requests import HTTPError
+from psaw import PushshiftAPI
+import praw
+import finviz
+from gamestonk_terminal import config_terminal as cfg
+
+from gamestonk_terminal.common.behavioural_analysis.reddit_helpers import find_tickers
+
+l_sub_reddits = [
+    "Superstonk",
+    "pennystocks",
+    "RobinHoodPennyStocks",
+    "Daytrading",
+    "StockMarket",
+    "stocks",
+    "investing",
+    "wallstreetbets",
+]
+
+# pylint:disable=inconsistent-return-statements
+
+
+def get_watchlists(
+    n_to_get: int,
+) -> Tuple[List[praw.models.reddit.submission.Submission], Dict, int]:
+    """Get reddit users watchlists
+
+    Parameters
+    ----------
+    n_to_get : int
+        Number of posts to look through
+
+    Returns
+    -------
+    List[praw.models.reddit.submission.Submission]:
+        List of reddit submissions
+    Dict:
+        Dictionary of tickers and counts
+    int
+        Count of how many posts were analyzed
+    """
+    d_watchlist_tickers: Dict = {}
+    l_watchlist_author = []
+    subs = []
+    try:
+        praw_api = praw.Reddit(
+            client_id=cfg.API_REDDIT_CLIENT_ID,
+            client_secret=cfg.API_REDDIT_CLIENT_SECRET,
+            username=cfg.API_REDDIT_USERNAME,
+            user_agent=cfg.API_REDDIT_USER_AGENT,
+            password=cfg.API_REDDIT_PASSWORD,
+        )
+    except ResponseException:
+        print(
```

Comment:
Same comment as above.  If someone is using this code only (not the terminal necessarily), then this catch tells them exactly what went wrong instead of just giving a large stack trace.

Reply:
The large stack trace should actually be even more helpful than this.

I wouldn't mind having this tbh, but I like consistency. And we haven't been doing this, so would be good to keep code consistent across database